Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPStreamSink.cpp

```diff
@@ -0,0 +1,433 @@
+    pOut[0] = (byte)0x80;                               // RTP version
+
+    if (bLastNalOfFrame)
+    {
+        pOut[1] = (byte)(h264payloadType | 1 << 7);  //last packet of frame marker bit
+    }
+    else
+    {
+        pOut[1] = (byte)(h264payloadType);
+    }
+    pOut[2] = m_SequenceNumber >> 8;
+    pOut[3] = m_SequenceNumber & 0x0FF;           // each packet is counted with a sequence counter
+    m_SequenceNumber++;
+    for (auto& ct : m_RtpStreamers)
+    {
+        auto ssrc = ct.second.m_ssrc;
+        auto ts1 = ts;
+        pOut[4] = (BYTE)((ts1 & 0xFF000000) >> 24);   // each image gets a timestamp
+        pOut[5] = (BYTE)((ts1 & 0x00FF0000) >> 16);
+        pOut[6] = (BYTE)((ts1 & 0x0000FF00) >> 8);
+        pOut[7] = (BYTE)((ts1 & 0x000000FF));
+
+        pOut[8] = (BYTE)((ssrc & 0xFF000000) >> 24);// 4 BYTE SSRC (sychronization source identifier)
+        pOut[9] = (BYTE)((ssrc & 0x00FF0000) >> 16);
+        pOut[10] = (BYTE)((ssrc & 0x0000FF00) >> 8);
+        pOut[11] = (BYTE)((ssrc & 0x000000FF));
+
+        ct.second.SendPacket(buf);
+    }
+}
+
+void RTPVideoStreamSink::PacketizeMode1(BYTE* bufIn, size_t szIn, LONGLONG llSampleTime)
+{
+    BYTE* sc = FindSC(bufIn, bufIn + szIn);
+    BYTE* pOut = m_pTxBuf.data();
+    size_t szOut = m_MTUSize;
+    BYTE* pOutEnd = pOut + szOut;
+    BYTE* sc1 = sc;
+    auto pOutCurr = pOut + rtpHeaderSize + stapAtypeHeaderSize;
+    uint32_t numNalsToSend = 0;
+    while (sc < (bufIn + szIn))
+    {
+        auto sc0 = sc;
+        while ((sc < bufIn + szIn) && !(*sc++));
+        sc1 = FindSC(sc, bufIn + szIn);
+        size_t nalsz = sc1 - sc;
+        if (pOutCurr + nalsz + stapASzHeaderSize > pOutEnd)
+        {
+            if (numNalsToSend == 0)
+            {
+                // STAP-A FU
+                pOut[rtpHeaderSize] = 28; // STAP-A FU type
+                pOut[rtpHeaderSize] |= (sc[0] & 0x60);
+                pOut[rtpHeaderSize + 1] = sc[0] & 0x1F;
+                pOut[rtpHeaderSize + 1] |= (1 << 7);
+                sc++; nalsz--;
+                auto maxSz = szOut - (rtpHeaderSize + stapAtypeHeaderSize + 1);
+                while (nalsz > maxSz)
+                {
+                    memcpy(&pOut[rtpHeaderSize + stapAtypeHeaderSize + 1], sc, maxSz);
+                    m_pTxBuf.Length((uint32_t)szOut);
+                    SendPacket(m_pTxBuf, llSampleTime, false);
+                    nalsz -= maxSz;
+                    sc += maxSz;
+                    pOut[rtpHeaderSize + 1] &= (~(1 << 7));
+                }
+
+                memcpy(&pOut[rtpHeaderSize + stapAtypeHeaderSize + 1], sc, nalsz);
+                pOut[rtpHeaderSize + 1] |= (1 << 6);
+                m_pTxBuf.Length(uint32_t(nalsz + rtpHeaderSize + stapAtypeHeaderSize + 1));
+                SendPacket(m_pTxBuf, llSampleTime, (sc1 == (bufIn + szIn)));
+                sc += nalsz;
+                sc = sc1;
+            }
+            //else if (numNalsToSend == 1)

```

Comment:
>//els [](start = 12, length = 5)

remove ?

Reply:
I am waiting on Anders to give me inputs on this corner case. As of now, it never hits

---
In reply to: [491104714](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491104714) [](ancestors = 491104714)